tests/cppunit/string_util_test.cc

```diff
@@ -268,3 +268,41 @@ TEST(StringUtil, RegexMatchExtractSSTFile) {
     ASSERT_TRUE(match_results[1] == "/000038.sst");
   }
 }
+
+TEST(StringUtil, SplitArguments) {
+  std::map<std::string, std::vector<std::string>> valid_cases = {
+      // normal cases
+      {"a b c", {"a", "b", "c"}},
+      {"a\tb\nc\fd", {"a", "b", "c", "d"}},
+
+      //  quote cases
+      {"hello \"a b\" c", {"hello", "a b", "c"}},
+      {"'a b' c", {"a b", "c"}},
+
+      {"\"a\\\"b\" c", {"a\"b", "c"}},
+      {"'a\\' b' c", {"a' b", "c"}},
```

Comment:
Could you add a test case like `command "a b" " c d "`?

Reply:
Sure, will have it later.

Reply:
Also we can use unescaped raw string `R"(...)"` in C++.

Reply:
@PragmaTwice Thanks for your tips. Will use the raw string to make it more clear.